### PR TITLE
ensure sources and mirror info is in stream list

### PR DESCRIPTION
**Pull request by ripienaar**
Signed-off-by: R.I.Pienaar <rip@devco.net>

/cc @nats-io/core

### CODE REVIEW

File: server/jetstream_cluster.go [hunk 1]
```diff
@@ -1549,7 +1549,7 @@ func (js *jetStream) processStreamLeaderChange(mset *stream, isLeader bool) {
 		resp.Error = jsError(err)
 		s.sendAPIErrResponse(client, acc, subject, reply, _EMPTY_, s.jsonResponse(&resp))
 	} else {
-		resp.StreamInfo = &StreamInfo{Created: mset.createdTime(), State: mset.state(), Config: mset.config(), Cluster: js.clusterInfo(mset.raftGroup())}
+		resp.StreamInfo = &StreamInfo{Created: mset.createdTime(), State: mset.state(), Config: mset.config(), Cluster: js.clusterInfo(mset.raftGroup()), Sources: mset.sourcesInfo(), Mirror: mset.mirrorInfo()}
```
Comment (ripienaar) [inline server/jetstream_cluster.go:1552]: not sure if this has to be here or not tbh

File: server/jetstream_cluster.go [hunk 2]
```diff
@@ -1831,7 +1831,7 @@ func (js *jetStream) processClusterUpdateStream(acc *Account, sa *streamAssignme
 		resp.Error = jsError(err)
 		s.sendAPIErrResponse(client, acc, subject, reply, _EMPTY_, s.jsonResponse(&resp))
 	} else {
-		resp.StreamInfo = &StreamInfo{Created: mset.createdTime(), State: mset.state(), Config: mset.config(), Cluster: js.clusterInfo(mset.raftGroup())}
+		resp.StreamInfo = &StreamInfo{Created: mset.createdTime(), State: mset.state(), Config: mset.config(), Cluster: js.clusterInfo(mset.raftGroup()), Mirror: mset.mirrorInfo(), Sources: mset.sourcesInfo()}
```
Comment (ripienaar) [inline server/jetstream_cluster.go:1834]: not sure if this has to be here or not tbh

Comment (derekcollison) [reply server/jetstream_cluster.go:1834]: Yes to this one.

File: server/jetstream_cluster.go (hunk 1 above)
Comment (derekcollison) [reply server/jetstream_cluster.go:1552]: yes needs to be here too.

**Review by derekcollison — APPROVED** [review]
LGTM

**Comment by ripienaar** [comment]
flapper tests, merging anyway

**Pull request merged**